src/observable/dom/WebSocketSubject.ts

```diff
@@ -46,37 +46,154 @@ export class WebSocketSubject<T> extends AnonymousSubject<T> {
+   * subscribes and unsubscribes. Server can use them to verify that some kind of messages should start or stop
+   * being forwarded to the client. In case of our application, after getting subscription message with proper identifier,
+   * gateway server can decide that it should connect to real sport news service and start forwarding messages from it.
+   * Note that both messages will be sent as returned by the functions, so you have to serialize them yourself, just
+   * as messages pushed via `next`. Also bear in mind that these messages will be sent on *every* subscription and
+   * unsubscription. This is potentially dangerous, because one consumer of an Observable may unsubscribe and the server
+   * might stop sending messages, since it got unsubscription message. You need to either handle this case
+   * on the server or use {@link publish} on a Observable returned from 'multiplex'.
    *
-   * @example <caption>Wraps WebSocket from nodejs-websocket (using node.js)</caption>
+   * Last argument to `multiplex` is a `messageFilter` function which filters out messages
+   * sent by the server to only those that belong to simulated WebSocket stream. For example server might mark these
+   * messages with some kind of string identifier on a message object and `messageFilter` would return `true`
+   * if there is such identifier on an object emitted by the socket.
    *
-   * import { w3cwebsocket } from 'websocket';
+   * Return value of `multiplex` is an Observable with messages incoming from emulated socket connection. Note that this
+   * is not a `WebSocketSubject`, so calling `next` or `multiplex` again will fail. For pushing values to the
+   * server you still need to use root `WebSocketSubject`.
    *
-   * let socket$ = Observable.webSocket({
-   *   url: 'ws://localhost:8081',
-   *   WebSocketCtor: w3cwebsocket
-   * });
    *
-   * socket$.subscribe(
-   *    (msg) => console.log('message received: ' + msg),
-   *    (err) => console.log(err),
-   *    () => console.log('complete')
+   * @example <caption>Listening for messages from the server.</caption>
+   * const subject = Rx.Observable.webSocket('ws://localhost:8081');
+   *
+   * subject.subscribe(
+   *    (msg) => console.log('message received: ' + msg), // Called whenever there is a message from the server.
+   *    (err) => console.log(err), // Called if at any point WebSocket API signals some kind of error.
+   *    () => console.log('complete') // Called when connection is closed (for whatever reason).
    *  );
    *
-   * socket$.next(JSON.stringify({ op: 'hello' }));
    *
-   * @param {string | WebSocketSubjectConfig} urlConfigOrSource the source of the websocket as an url or a structure defining the websocket object
-   * @return {WebSocketSubject}
+   * @example <caption>Pushing messages to the server.</caption>
+   * const subject = Rx.Observable.webSocket('ws://localhost:8081');
```

Comment:
As a matter of fact, it is still off, since I subscribe and complete immediately after, which I guess means nothing will be sent anyways.

But to make it 100% legit I would have to use openObserver, which would make example super convoluted for newcomers, so for educational reasons I left it as is, hoping comments are clear enough